src/Nethermind/Nethermind.Tools.Kute/Metrics.cs

```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Tools.Kute;
+
+public class Metrics
+{
+    public int Messages { get; private set; }
+    public int Failed { get; private set; }
+    public int Ignored { get; private set; }
+    public int Responses { get; private set; }
+    public IDictionary<string, int> Requests { get; } = new Dictionary<string, int>();
+
+    public TimeSpan TotalRunningTime { get; set; }
+
+    public void TickMessages() => Messages++;
+    public void TickFailed() => Failed++;
+    public void TickIgnored() => Ignored++;
+    public void TickResponses() => Responses++;
+
+    public void TickMethod(string methodName)
+    {
+        if (Requests.ContainsKey(methodName))
+        {
+            Requests[methodName]++;
+        }
+        else
+        {
+            Requests[methodName] = 0;
+        }
```

Comment:
Is there an other way to do the "modify key, and if it does not exist, initialize to a default value"?